From b21e42103b1a01a45b4058c06f3c9d5e41d169c4 Mon Sep 17 00:00:00 2001
From: Nick Timkovich <npt@uchicago.edu>
Date: Fri, 4 Aug 2017 12:56:19 -0500
Subject: [PATCH 6/6] Use specified phrases in lieu of 'Used' for limit summary

Some resources, namely floating IPs, are limited based on those
allocated, rather than in actual use. The "Allocated" string is present
in the code but not used in the template. This effectively changes "Used
X of Y" to "Allocated X of Y" for floating IPs.

Change-Id: I7919b883a47d66b1e870862363d07e0f90bdbe7b
Closes-Bug: #1708733
---
 horizon/templates/horizon/common/_limit_summary.html | 16 ++++++++--------
 openstack_dashboard/usage/views.py                   |  6 ++++--
 2 files changed, 12 insertions(+), 10 deletions(-)

diff --git a/horizon/templates/horizon/common/_limit_summary.html b/horizon/templates/horizon/common/_limit_summary.html
index b58ba8b4c..12c441d3a 100644
--- a/horizon/templates/horizon/common/_limit_summary.html
+++ b/horizon/templates/horizon/common/_limit_summary.html
@@ -13,21 +13,21 @@
           <div class="quota_subtitle">
             {% if quota.max|quotainf != '-1' %}
               {% if quota.type == "totalRAMUsed" %}
-                {% blocktrans trimmed with used=quota.used|mb_float_format available=quota.max|quotainf|mb_float_format %}
-                  Used <span> {{ used }} </span>of<span> {{ available }} </span>
+                {% blocktrans trimmed with usedphrase=quota.text used=quota.used|mb_float_format available=quota.max|quotainf|mb_float_format %}
+                  {{ usedphrase }} <span> {{ used }} </span>of<span> {{ available }} </span>
                 {% endblocktrans %}
               {% elif quota.type == "totalGigabytesUsed" %}
-                {% blocktrans trimmed with used=quota.used|diskgbformat available=quota.max|quotainf|diskgbformat %}
-                  Used <span> {{ used }} </span>of<span> {{ available }} </span>
+                {% blocktrans trimmed with usedphrase=quota.text used=quota.used|diskgbformat available=quota.max|quotainf|diskgbformat %}
+                  {{ usedphrase }} <span> {{ used }} </span>of<span> {{ available }} </span>
                 {% endblocktrans %}
               {% else %}
-                {% blocktrans trimmed with used=quota.used|intcomma available=quota.max|quotainf|intcomma %}
-                  Used <span> {{ used }} </span>of<span> {{ available }} </span>
+                {% blocktrans trimmed with usedphrase=quota.text used=quota.used|intcomma available=quota.max|quotainf|intcomma %}
+                  {{ usedphrase }} <span> {{ used }} </span>of<span> {{ available }} </span>
                 {% endblocktrans %}
               {% endif %}
             {% else %}
-              {% blocktrans trimmed with used=quota.used|intcomma %}
-                Used <span> {{ used }} </span>(No Limit)
+              {% blocktrans trimmed with usedphrase=quota.text used=quota.used|intcomma %}
+                {{ usedphrase }} <span> {{ used }} </span>(No Limit)
               {% endblocktrans %}
             {% endif %}
           </div>
diff --git a/openstack_dashboard/usage/views.py b/openstack_dashboard/usage/views.py
index 23a6ece5a..35e654311 100644
--- a/openstack_dashboard/usage/views.py
+++ b/openstack_dashboard/usage/views.py
@@ -10,6 +10,7 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+from django.utils.translation import pgettext_lazy
 from django.utils.translation import ugettext_lazy as _
 
 from horizon import exceptions
@@ -68,7 +69,8 @@ class UsageView(tables.DataTableView):
                  #("totalCoresUsed", "maxTotalCores", _("VCPUs")),
                  #("totalRAMUsed", "maxTotalRAMSize", _("RAM")),
                  ("totalFloatingIpsUsed", "maxTotalFloatingIps",
-                  _("Floating IPs"), _("Allocated")),
+                  _("Floating IPs"),
+                  pgettext_lazy('Label in the limit summary', "Allocated")),
                  ("totalSecurityGroupsUsed", "maxSecurityGroups",
                   _("Security Groups"))]
         # Check for volume usage
@@ -80,7 +82,7 @@ class UsageView(tables.DataTableView):
                          _("Volume Storage")))
         for t in types:
             if t[0] in self.usage.limits and t[1] in self.usage.limits:
-                text = False
+                text = pgettext_lazy('Label in the limit summary', 'Used')
                 if len(t) > 3:
                     text = t[3]
                 context['charts'].append({
-- 
2.15.0

